lib/engine/action/place_token.rb

```diff
@@ -5,23 +5,26 @@
 module Engine
   module Action
     class PlaceToken < Base
-      attr_reader :city, :slot
+      attr_reader :city, :slot, :tokenidx
 
-      def initialize(entity, city, slot)
+      def initialize(entity, city, slot, tokenidx = nil)
         @entity = entity
         @city = city
         @slot = slot
+        @tokenidx = tokenidx
       end
 
       def self.h_to_args(h, game)
-        [game.city_by_id(h['city']), h['slot']]
+        [game.city_by_id(h['city']), h['slot'], h['tokenidx']]
       end
 
       def args_to_h
-        {
+        h = {
```

Comment:
you can just add token_id in here, it will be cleared automatically if nil (please test that is true)

Reply:
yes it is.